openshift/dynamic/client.py

```diff
@@ -0,0 +1,394 @@
+#!/usr/bin/env python
+
+import sys
+import json
+from functools import partial
+
+import yaml
+from pprint import pformat
+
+from kubernetes import config
+from kubernetes.client.api_client import ApiClient
+from kubernetes.client.rest import ApiException
+
+
+class DynamicClient(object):
+
+    def __init__(self, client):
+        self.client = client
+        self.__resources = ResourceContainer(self.parse_api_groups())
+
+    @property
+    def resources(self):
+        return self.__resources
+
+    def default_groups(self):
+        groups = {}
+        groups['api'] = { '': {
+            'v1': self.get_resources_for_api_version('api', '', 'v1', True)
+        }}
+
+        try:
+            self.request('get', '/version/openshift')
+            is_openshift = True
+        except ApiException:
+            is_openshift = False
+
+        if is_openshift:
+            groups['oapi'] = { '': {
```

Comment:
Can you not just move this under the `self.request('get', '/version/openshift')`? Like:

```
        try:
            self.request('get', '/version/openshift')
            groups['oapi'] = { '': {
                'v1': self.get_resources_for_api_version('oapi', '', 'v1', True)
            }}
        except ApiException:
            pass
```

Reply:
Good catch, refactor cruft